apps/balancer-tools/src/app/apr/api/(utils)/fetchForDateRange.ts

```diff
@@ -1,142 +1,67 @@
-      skip,
-      createdBefore: createdBefore,
-      limit,
-      minTvl,
-      maxTvl,
-      block,
-    });
-  } catch (e) {
-    // If this errors out, probably the subgraph hadn't been deployed yet at this block
-    return [];
-  }
-  let fetchedPools: typeof response.pools;
-  console.log(
-    `Fetched ${response.pools.length} pools from ${network}(${networkIdFor(
-      network,
-    )}) for block ${block}`,
-  );
-  fetchedPools = response.pools;
-
-  if (response.pools.length > limit) {
-    fetchedPools = [
-      ...fetchedPools,
-      ...(await fetchPoolsFromNetwork(
-        network,
-        endAt,
-        maxTvl,
-        minTvl,
-        skip + limit,
-      )),
-    ];
-  }
-  return fetchedPools;
-};
-
-const fetchPools = async (
-  network: string,
+export async function fetchDataForDateRange(
+  startDate: Date,
   endDate: Date,
-  maxTvl = 10_000_000_000,
-  minTvl = 10_000,
-) => {
-  const networks = network
-    ? [network]
-    : Object.values(Network).filter(
-        (network) => network !== Network.Sepolia && network !== Network.Goerli,
-      );
-  const allFetchedPools = await Promise.all(
-    networks.map(
-      async (network) =>
-        await fetchPoolsFromNetwork(network, endDate, maxTvl, minTvl),
-    ),
-  );
+  maxTvl: number = 10_000_000_000,
+  minTvl: number = 10_000,
+): Promise<{ [key: string]: PoolStatsData[] }> {
+  const existingPoolForDate = await db.execute(sql`
+    SELECT
+      p.external_id AS id,
+      p.symbol AS symbol,
+      p.pool_type AS pool_type,
+      p.network_slug AS network
+    FROM
+      pool_snapshots ps
+      JOIN pools p ON p.external_id = ps.pool_external_id
+    WHERE
+      ps.timestamp = '${sql.raw(endDate.toISOString())}'
+      AND ps.liquidity > ${sql.raw(minTvl.toString())}
+      AND ps.liquidity < ${sql.raw(maxTvl.toString())}
+  `);
+  console.log(`fetched ${existingPoolForDate.length} pools`);
+  const perDayData: { [key: string]: PoolStatsData[] } = {};
 
-  // Flatten the array of arrays into a single array.
-  return allFetchedPools.flat();
-};
-export const fetchDataForDateRange = withCache(
-  async function fetchDataForDateRangeFn(
-    startDate: Date,
-    endDate: Date,
-    network: string,
-    maxTvl: number = 10_000_000_000,
-    minTvl: number = 10_000,
-  ): Promise<{ [key: string]: PoolStatsData[] }> {
-    const existingPoolForDate = await fetchPools(
-      network,
-      endDate,
-      maxTvl,
-      minTvl,
-    );
-    console.log(`fetched ${existingPoolForDate.length} pools`);
-    const perDayData: { [key: string]: PoolStatsData[] } = {};
+  await Promise.all(
+    existingPoolForDate.map(async (pool) => {
+      let gaugesData;
+      try {
+        gaugesData = await fetcher<PoolStatsResults>(
+          `${BASE_URL}/apr/api?startAt=${formatDateToMMDDYYYY(
+            startDate,
+          )}&endAt=${formatDateToMMDDYYYY(endDate)}&poolId=${pool.id}`,
+        );
```

Comment:
we could theoretically get rid of this now, and it also skews our results a bit given our new reality. have you discussed this with Fábio?

Reply:
Yes! we discussed and we think it makes more sense to show the average of the days instead of the apr of the range date. What we have to do is to make that more clear to the user

Reply:
AA I think now I got what you mean, we don't have the add one layer of API, we can just call the query directly